lib/util.js

```diff
@@ -345,11 +349,35 @@ function getSystemErrorName(err) {
   return internalErrorName(err);
 }
 
+function addSafeAbortSignalAbortListener(signal, listener, options = kEmptyObject) {
+  if (signal === undefined) {
+    throw new ERR_INVALID_ARG_TYPE('signal', 'AbortSignal', signal);
+  }
+  validateAbortSignal(signal, 'signal');
+  kResistStopPropagation ??= require('internal/event_target').kResistStopPropagation;
+  // TODO(atlowChemi) add { subscription: true } and return directly
+  signal.addEventListener(
+    'abort',
+    listener,
+    { __proto__: null, ...options, once: true, [kResistStopPropagation]: true },
```

Comment:
I think it makes sense to enforce the `once` option for the `abort` event, but I'd be happy to know if anyone thinks otherwise I'm happy to change 🙂

Reply:
`once: true` makes sense for me here (and also not accepting options).

I'm wondering if we should do

```js
if (signal.aborted) {
  queueMicrotask(() => listener())` 
}
```

(and either way the alternative is to at least check for signal.aborted since the behavior right now would be to add the listener and not do cleanup)

Reply:
OK, I re-implemented, let me know what you think 🙂

Reply:
> I was thinking of exposing a "safe" version of addEventListener.

Forgetting to cleanup if the signal is already aborted is a common bug and if we can help users with it that's good.

We can still use kWeakRef I believe? The idea of kWeakListener (or whatever we named it) is to prevent cases where the user forgets to remove the listener and the listener keeps a live reference to the object.  I'm even less sure if we should expose it.

Reply:
Oh I was not talking of exposing it, just about allowing us to use it via this utility, in internal calls